### PR TITLE
fix-multinomial-logit-dist

### LINKED ISSUE #529: Bug in multinomial distribution wrt to softmax^-1
This was reported on Twitter https://twitter.com/clamusgh/status/1526397666527064064

image

### PR DESCRIPTION
**Pull request by spinkney**
#### Submission Checklist

- [ ] Builds locally 
- [x] Declare copyright holder and open-source license: see below

#### Summary

I've declared an different variable \gamma as the `log_softmax(theta)`. I'd like @bob-carpenter to weigh in and see if this is more clear. 

Screen Shot 2022-05-18 at 3 08 33 PM

fixes #529 

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company): Sean Pinkney

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: src/functions-reference/multivariate_discrete_distributions.Rmd [hunk 1]
```diff
@@ -68,48 +68,48 @@ scale.
...
 ### Probability mass function
 
-If $K \in \mathbb{N}$, $N \in \mathbb{N}$, and $\text{softmax}^{-1}(\theta) \in
+If $K \in \mathbb{N}$, $N \in \mathbb{N}$, and $\text{softmax}(\theta) \in
 \text{$K$-simplex}$, then for $y \in \mathbb{N}^K$ such that
 $\sum_{k=1}^K y_k = N$, \[
-\text{MultinomialLogit}(y|\theta) = \text{Multinomial}(y|\text{softmax}^{-1}(\theta)) =
-\binom{N}{y_1,\ldots,y_K} \prod_{k=1}^K [\text{softmax}^{-1}(\theta)_k]^{y_k}, \] where the
+\text{MultinomialLogit}(y \mid \gamma = \text{log\_softmax}(\theta)) = \text{Multinomial}(y \mid \text{softmax}(\theta)) =
+\binom{N}{y_1,\ldots,y_K} \prod_{k=1}^K [\text{softmax}(\theta)_k]^{y_k}, \] where the
```
Comment (bob-carpenter) [inline src/functions-reference/multivariate_discrete_distributions.Rmd:75]: Line 74 should just be:

```
MultinomialLogit(y | gamma)
= Multinomial(y | softmax(gamma))
= ... formula ...
```

The only problem with the original was a stray `^{-1}`.

File: src/functions-reference/multivariate_discrete_distributions.Rmd [hunk 2]
```diff
@@ -68,48 +68,48 @@ scale.
...
-<!-- array[] int; multinomial_logit_rng; (vector theta, int N); -->
-\index{{\tt \bfseries multinomial\_logit\_rng }!{\tt (vector theta, int N): array[] int}|hyperpage}
+<!-- array[] int; multinomial_logit_rng; (vector gamma, int N); -->
+\index{{\tt \bfseries multinomial\_logit\_rng }!{\tt (vector gamma, int N): array[] int}|hyperpage}
 
-`array[] int` **`multinomial_logit_rng`**`(vector theta, int N)`<br>\newline
-Generate a multinomial variate with simplex distribution parameter
-$\text{softmax}^{-1}(\theta)$ and total count $N$; may only be used in transformed data and
+`array[] int` **`multinomial_logit_rng`**`(vector gamma, int N)`<br>\newline
+Generate a multinomial variate with log simplex distribution parameter
```
Comment (bob-carpenter) [inline src/functions-reference/multivariate_discrete_distributions.Rmd:112]: The first one would be right without the `^{-1}`.  The second one could be confusing because `softmax` isn't invertible.  How about the following.

```
Generate a variate from a multinomial distribution with probabilities `softmax(gamma)` and total count `N`.
```

This works around the problem of softmax not being invertible.  I do like changing the variate to `gamma` to not confuse it with the `theta` we tend to use for simplexes.

Also, it needs a final period (this was an earlier problem apparently).

**Review by bob-carpenter — CHANGES_REQUESTED** [review]
thanks for updating!  just a couple small changes

**Review by bob-carpenter — APPROVED** [review]
Thanks for the update.  Looks great.
